server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -149,7 +152,27 @@ object InteractiveSession extends Logging {
     builderProperties ++= conf
 
     def livyJars(livyConf: LivyConf, scalaVersion: String): List[String] = {
-      Option(livyConf.get(LIVY_REPL_JARS)).map(_.split(",").toList).getOrElse {
+      Option(livyConf.get(LIVY_REPL_JARS)).map { jars =>
+        val regex = """[\w-]+_(\d\.\d\d).*\.jar""".r
+        jars.split(",").flatMap { n =>
+          val trimmed = n.trim
+          if (!trimmed.startsWith("local")) {
+            // expand the glob path for file and hdfs scheme
+            val path = new Path(toURI(trimmed))
+            val pathFs = FileSystem.get(path.toUri, livyConf.hadoopConf)
```

Comment:
Do you mind adding a simple check to bypass glob expansion if glob patterns are absent?
Or we could also remove glob expansion entirely?